src/snowflake/snowpark/_internal/code_generation.py

```diff
@@ -0,0 +1,594 @@
+# {name} is of type {type(obj)} and serialized by snowpark-python
+"""
+                except Exception as exc:
+                    logger.debug(
+                        f"Unable to generate source code for object {name} of type {type(obj)} due to exception {exc}"
+                    )
+                    raise
+    return func_text, global_vars_text
+
+
+def resolve_target_func_imports(
+    to_import: Set[Tuple[str, str]],
+    to_import_from_module: Dict[str, Set[Tuple[str, str]]],
+) -> str:
+    """
+    Deal with imports and alias, generate imports string.
+
+    Args:
+        to_import: set of name and alias pairs of direct imports which should be generated as "import xxx"
+            or "import xxx as yyy".
+        to_import_from_module: dict of import information, key is the module name with value being the set of
+            name and alias paris of imported objects which should be generated as "from xxx import yyy" or
+            "from xxx import yyy as zzz".
+
+    Returns:
+        A string of generated imports.
+    """
+    imports = [
+        f"import {name + ' as ' if name != alias else ''}{alias}"
+        for name, alias in sorted(to_import)
+    ]
+    for module, name_alias_pairs in sorted(to_import_from_module.items()):
+        classes = ", ".join(
+            f"{name + ' as ' if name != alias else ''}{alias}"
+            for name, alias in sorted(name_alias_pairs)
+        )
+        imports.append(f"from {module} import {classes}")
+    return "\n".join(imports) + ("\n" if imports else "")
+
+
+def handle_target_func_self_source_code(
+    func: FunctionType, source_code_without_target_func: str, code_as_comment: bool
+) -> Tuple[str, str]:
+    """
+    Generate the source code of the target func itself and apply function assignment.
+
+    Args:
+        func: The target function to generate source code for.
+        source_code_without_target_func: The generated code without the target function. The target function code
+            and function assignment will be appended to this one.
+        code_as_comment: Whether the code will be generated as comment.
+
+    Returns:
+        A tuple of two strings, the first one is the complete source code including target functions and all of its
+        referenced objects, and the second one is function assignment.
+    """
+    func_module_name = func.__module__
+    complete_source_code = source_code_without_target_func
+    if isinstance(func, FunctionType):
+        func_source_code = textwrap.dedent(inspect.getsource(func))
+        if not code_as_comment:
+            func_source_code = remove_function_udf_annotation(func_source_code)
+        if not is_lambda(func):
+            complete_source_code = f"{complete_source_code}\n{func_source_code}"
+        func_assignment = (
+            get_lambda_code_text(func_source_code) if is_lambda(func) else func.__name__
+        )
+    else:
+        # BuiltinFunctionType
```

Comment:
We might want to claim this in `isinstance` and raise for `else`?

Reply:
it is already checked at the beginning of `generate_source_code` via `check_func_type`. I could add a comment in this func.

Reply:
Right. I guess I'm just a bit paranoid and feels it is always good to explicitly check our assumption lol, so that if one day there is a new type, we won't have something goes wrong silently :)

Reply:
make sense to me, I've updated the code